crates/extrinsics/src/lib.rs

```diff
@@ -303,6 +303,44 @@ pub fn parse_code_hash(input: &str) -> Result<<DefaultConfig as Config>::Hash> {
     Ok(arr.into())
 }
 
+/// Calculate the total contract deposit.
+async fn get_contract_total_deposit(
+    storage_base_deposit: Balance,
+    storage_item_deposit: Balance,
+    storage_byte_deposit: Balance,
+    client: &Client,
+) -> Result<Balance> {
+    let contract_pallet_version = fetch_contracts_pallet_version(client).await?;
+    let mut contract_deposit = storage_base_deposit
+        .saturating_add(storage_item_deposit)
+        .saturating_add(storage_byte_deposit);
+
+    // From contracts pallet version 10 deposit calculation has changed.
+    if contract_pallet_version >= 10 {
```

Comment:
Need to display an error to the user otherwise

Reply:
For the pallet-contracts version >=8, it works. The issue may arise for pallets-contract versions <8, as it will result in an error during the scale decoding of ContractInfo struct, which has changed at this time. It's important to note that this limitation is not introduced by this commit.